src/Teams/beta/custom/SetMgBetaChatRscConfiguration_Update.cs

```diff
@@ -0,0 +1,380 @@
+                }
+                // get the client instance
+                try
+                {
+                    await ((Microsoft.Graph.Beta.PowerShell.Runtime.IEventListener)this).Signal(Microsoft.Graph.Beta.PowerShell.Runtime.Events.CmdletBeforeAPICall); if (((Microsoft.Graph.Beta.PowerShell.Runtime.IEventListener)this).Token.IsCancellationRequested) { return; }
+
+                    MGTeamsInternalAuthorizationPolicy authorizationPolicy = await this.Client.GetAuthorizationPolicy(eventListener: this, sender: Pipeline);
+                    WriteVerbose($"PermissionGrantPolicies currently assigned to default user role: '{string.Join(", ", authorizationPolicy.DefaultUserRolePermissions.PermissionGrantPoliciesAssigned)}'.");
+
+                    if (this.State == MicrosoftGraphRscConfigurationState.Disabled)
+                    {
+                        // Disable chat RSC Teams Setting.
+                        await this.Client.UpdateTeamsAppSettings(
+                            isChatResourceSpecificConsentEnabled: false,
+                            eventListener: this,
+                            sender: Pipeline);
+
+                        WriteVerbose($"Disabled Chat RSC Teams setting.");
+
+                        if (((Microsoft.Graph.Beta.PowerShell.Runtime.IEventListener)this).Token.IsCancellationRequested) { return; }
+
+                        // Disable preapproval configs.
+                        if (authorizationPolicy.DefaultUserRolePermissions.PermissionGrantPoliciesAssigned.Contains(RscConfigurationConverter.MicrosoftCreatedPermissionGrantPolicyForChatRscPreApproval, StringComparer.OrdinalIgnoreCase))
+                        {
+                            IEnumerable<string> updatedPermissionGrantPolicies = authorizationPolicy.DefaultUserRolePermissions.PermissionGrantPoliciesAssigned.Except(
+                                new string[] { RscConfigurationConverter.MicrosoftCreatedPermissionGrantPolicyForChatRscPreApproval },
+                                StringComparer.OrdinalIgnoreCase);
+                            await this.Client.UpdateDefaultUserRolePermissionGrantPoliciesAssigned(
+                                updatedPermissionGrantPolicies,
+                                this,
+                                Pipeline);
+
+                            WriteVerbose($"Updated permission grant policies assigned to default user role: '{string.Join(", ", updatedPermissionGrantPolicies)}'.");
+                        }
+
+                        if (((Microsoft.Graph.Beta.PowerShell.Runtime.IEventListener)this).Token.IsCancellationRequested) { return; }
+                    }
+                    else if (this.State == MicrosoftGraphRscConfigurationState.EnabledForPreApprovedAppsOnly)
+                    {
+                        // Disable chat RSC Teams Setting.
```

Comment:
Let's flip the order. First add preapproval policy and then mark the global setting as disabled.